espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/UiautomatorParams.java

```diff
@@ -0,0 +1,166 @@
+package io.appium.espressoserver.lib.model;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+public class UiautomatorParams extends AppiumParams {
+    private Strategy strategy;
+    private String value;
+    private Integer index;
+    private Action action;
+
+    public Action getAction() {
+        return action;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    public Strategy getStrategy() {
+        return strategy;
+    }
+
+    @Nullable
+    public Integer getIndex() {
+        return index;
+    }
+
+    public enum Strategy {
+        @SerializedName("clazz")
+        CLASS_NAME("clazz"),
+
+        @SerializedName("res")
+        ID("res"),
+
+        @SerializedName("text")
+        TEXT("text"),
+
+        @SerializedName("textContains")
+        TEXT_CONTAIN("textContains"),
+
+        @SerializedName("textEndsWith")
+        TEXT_ENDS_WITH("textEndsWith"),
+
+        @SerializedName("textStartsWith")
+        TEXT_STARTS_WITH("textStartsWith"),
+
+        @SerializedName("desc")
+        DESC("desc"),
+
+        @SerializedName("descContains")
+        DESC_CONTAINS("descContains"),
+
+        @SerializedName("descEndsWith")
+        DESC_ENDS_WITH("descEndsWith"),
+
+        @SerializedName("descStartsWith")
+        DESC_STARTS_WITH("descStartsWith"),
+
+        @SerializedName("pkg")
+        APPLICATION_PACKAGE("pkg"),;
+
+        private final String name;
+
+        Strategy(final String name) {
+            this.name = name;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public static List<String> getValidStrategyNames() {
+            List<String> validStrategies = new ArrayList<>();
+            for (Strategy strategy : Strategy.values()) {
+                validStrategies.add(strategy.getName());
+            }
+            return validStrategies;
+        }
+    }
+
+
+    public enum Action {
+        @SerializedName("click")
+        CLICK("click"),
+
+        @SerializedName("longClick")
+        LONG_CLICK("longClick"),
+
+        @SerializedName("getText")
+        GET_TEXT("getText"),
+
+        @SerializedName("getContentDescription")
+        GET_CONTENT_DESCRIPTION("getContentDescription"),
+
+        @SerializedName("getClassName")
+        GET_CLASS_NAME("getClassName"),
+
+        @SerializedName("getResourceName")
+        GET_RESOURCE_NAME("getResourceName"),
+
+        @SerializedName("getVisibleBounds")
+        GET_VISIBLE_BOUNDS("getVisibleBounds"),
```

Comment:
is the resulting Rect properly serializable?

Reply:
Yes